### ISSUE TITLE
help needed with tf.reduce_mean()

### ISSUE THREAD

**Issue by aswanthkrishna** [tensorflow/quantum#273]
I have a 2D tensor of circuits. I wish to find the expectation values of an operator for these circuits, and take the mean along an axis. 
 I have tried using `tf.reduce_mean().` 

it is giving me this error.
>programs must be rank 1. Got rank 2. [Op:TfqSimulateExpectation]

```
#my model
result = tf.reduce_mean(
         tfq.layers.Expectation()(circuits,
                                  symbol_names = symbols,
                                  operators = Paulis,
                                  initializer=tf.keras.initializers.RandomUniform(0, 2 * np.pi)),axis = 1)
```
am I doing something wrong?
Thanks in advance!

**Comment by zaqqwerty** [comment]
Hi, our circuit execution layers currently accept either a single Cirq circuit or a 1D list or tensor of circuits.  To emulate higher dimensional circuit inputs, you can use the [tf.reshape](https://www.tensorflow.org/api_docs/python/tf/reshape) op on the output of our quantum layers.

Here is a code example to demonstrate the reshaping:
```
q = cirq.GridQubit(0, 0)
circuit_list_2d = [[cirq.Circuit(cirq.X(q)), cirq.Circuit(cirq.H(q))],
                   [cirq.Circuit(cirq.Y(q)), cirq.Circuit(cirq.S(q))]]
# Flatten to 1D
flat_circuit_list = []
for sublist in circuit_list_2d:
    flat_circuit_list += sublist
expectations_1d = tfq.layers.Expectation()(
    flat_circuit_list, symbol_names=[], operators=cirq.Z(q))
print(expectations)

# Expand back to 2D
expectations_2d = tf.reshape(expectations_1d, [len(circuit_list_2d), len(circuit_list_2d[0])])
print(expectations_2d)

result = tf.reduce_mean(expectations_2d, axis=1)
print(result)
```

**Comment by aswanthkrishna** [comment]
Thank you @zaqqwerty . can i train such a model?
